lib/main.go

```diff
@@ -214,7 +215,33 @@ func (obj *Main) Run() error {
 		return fmt.Errorf("hostname cannot be empty")
 	}
 
+	// Get current user running mgmt
+	user, _ := user.Current()
+
+	// Use /var/lib/mgmt if user is root, otherwise
+	// get systemd state directory if set, if not then
+	// use XDG_CACHE_DIR or default for XDG_CACHE_DIR
 	var prefix = fmt.Sprintf("/var/lib/%s/", obj.Program) // default prefix
+	stateDir := os.Getenv("STATE_DIRECTORY")
+	xdg := os.Getenv("XDG_CACHE_HOME")
+
+	if user.Uid == "0" {
```

Comment:
I think this logic can be refactored to be less nested and more straight forward... Want to give it a try?

Reply:
Ok, I'm glad you said that. I didn't like it but didn't put enough thought into it. I think this is better, is this more along the lines of what you were looking for? I'll rebase and squash into a single commit once you like everything.